service/collector/endpoint.go

```diff
@@ -47,23 +61,212 @@ func NewEndpoint(config EndpointConfig) (*Endpoint, error) {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
 	}
 
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		MaxIdleConns:    maxIdleConnection,
+	}
+
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   maxTimeoutSec * time.Second,
+	}
+
 	i := &Endpoint{
-		g8sClient: config.G8sClient,
-		k8sClient: config.K8sClient,
-		logger:    config.Logger,
+		g8sClient:     config.G8sClient,
+		httpClient:    client,
+		httpTransport: tr,
+		k8sClient:     config.K8sClient,
+		localIP:       getLocalIP(),
+		logger:        config.Logger,
 	}
 
 	return i, nil
 }
 
-func (i *Endpoint) Collect(ch chan<- prometheus.Metric) error {
+func (e *Endpoint) Collect(ch chan<- prometheus.Metric) error {
 	// TODO
 	// Here we will put implementation of collecting data for each endpoint IP
+	listOpts := metav1.ListOptions{}
+	getOpts := metav1.GetOptions{}
+
+	kvmConfigs, err := e.g8sClient.ProviderV1alpha1().KVMConfigs(crdNamespace).List(listOpts)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for _, kvmConfig := range kvmConfigs.Items {
+		if kvmConfig.DeletionTimestamp != nil {
+			// ignore clusters that are marked for deletion
+			continue
+		}
+
+		endpoint, err := e.k8sClient.CoreV1().Endpoints(kvmConfig.Name).Get(workerEndpoint, getOpts)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		ipList := getEndpointIps(endpoint)
+		for _, ip := range ipList {
```

Comment:
I'd maybe extract this loop to be in separate function for easier comprehension.